control/controlclient/noise.go

```diff
@@ -34,6 +37,57 @@ type noiseConn struct {
 	id   int
 	pool *noiseClient
 	h2cc *http2.ClientConn
+
+	readHeaderOnce    sync.Once     // guards init of reader field
+	reader            io.Reader     // (effectively Conn.Reader after header)
+	earlyPayloadReady chan struct{} // closed after earlyPayload is set (including set to nil)
+	earlyPayload      *tailcfg.EarlyNoise
+}
+
+func (c *noiseConn) RoundTrip(r *http.Request) (*http.Response, error) {
+	return c.h2cc.RoundTrip(r)
+}
+
+const hdrLen = 9 // http2 frame header size; also size of our early payload size header
```

Comment:
define this closer to earlyPayloadMagic, or move that up here.

Reply:
done w/ comments